modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -348,6 +384,45 @@ private String _getPropertyName(Object object) {
 		return String.valueOf(object);
 	}
 
+	private Object _getSub(Object left, Object right)
+		throws ExpressionVisitException {
+
+		if ((left instanceof String) && (right instanceof Duration)) {
+			try {
+				Date date = null;
+
+				if (Objects.equals(
```

Comment:
The new Date, should come from the evaluation of the right expression, when this is a NOW, not in the SUB.

Reply:
I implemented it and I developed the visit of the method Now() to return the NOW string. So, for that reason I am expecting here the NOW String. See:  https://github.com/brianchandotcom/liferay-portal/pull/128594/commits/78ba96fe7bfee3776ef203d3533628756fd0c54d

If it is wrong maybe we can change the other side as well.

Reply:
I don't now, but better if better is a type, instead a String NOW, because if the user write now, then the sub expression will work as well, right?